table/merging_iterator.cc

```diff
@@ -191,28 +227,35 @@ class MergingIterator : public InternalIterator {
       // just after the if-block.
       ClearHeaps();
       InitMaxHeap();
+      Slice target = key();
       for (auto& child : children_) {
         if (&child != current_) {
           if (!prefix_seek_mode_) {
-            child.Seek(key());
+            child.Seek(target);
             if (child.Valid()) {
               // Child is at first entry >= key().  Step back one to be < key()
               TEST_SYNC_POINT_CALLBACK("MergeIterator::Prev:BeforePrev",
                                        &child);
+              assert(child.status().ok());
               child.Prev();
             } else {
               // Child has no entries >= key().  Position at last entry.
               TEST_SYNC_POINT("MergeIterator::Prev:BeforeSeekToLast");
+              considerStatus(child.status());
               child.SeekToLast();
             }
+            considerStatus(child.status());
```

Comment:
I didn't find where the merging iterator is invalidated.

Reply:
Oh, this commit doesn't invalidate iterator, only makes `status()` more correct and fast. The next commit makes `Valid()` return false if status is not ok.